codegen/cmd_test.go

```diff
@@ -40,7 +41,103 @@ var _ = Describe("Cmd", func() {
 	skv2Imports.External["github.com/solo-io/cue"] = []string{
 		"encoding/protobuf/cue/cue.proto",
 	}
+	It("install conditional sidecars", func() {
+		var (
+			agentConditional = "and ($.Values.glooAgent.enabled) ($.Values.glooAgent.runAsSidecar)"
+		)
+
+		cmd := &Command{
+			Chart: &Chart{
+				Operators: []Operator{
+					{
+						Name: "gloo-mgmt-server",
+						Service: Service{
+							Ports: []ServicePort{{
+								Name:        "grpc",
+								DefaultPort: 9900,
+							}},
+						},
+						Rbac: []rbacv1.PolicyRule{{
+							Verbs:     []string{"*"},
+							APIGroups: []string{"coordination.k8s.io"},
+							Resources: []string{"leases"},
+						}},
+						Deployment: Deployment{
+							Sidecars: []Sidecar{{
+								Name: "gloo-agent",
+								Volumes: []v1.Volume{{
+									Name: "agent-volume",
+									VolumeSource: v1.VolumeSource{
+										Secret: &v1.SecretVolumeSource{
+											SecretName: "agent-volume",
+										},
+									},
+								}},
+								Rbac: []rbacv1.PolicyRule{{
+									Verbs:     []string{"*"},
+									APIGroups: []string{"apiextensions.k8s.io"},
+									Resources: []string{"customresourcedefinitions"},
+								}},
+								Container: Container{
+									Image: Image{
+										Registry:   "gcr.io/gloo-mesh",
+										Repository: "gloo-mesh-agent",
+										Tag:        "0.0.1",
+									},
+								},
+								Service: Service{
+									Ports: []ServicePort{{
+										Name:        "grpc",
+										DefaultPort: 9977,
+									}},
+								},
+								EnableStatement: agentConditional,
+							}},
+							Container: Container{
+								Image: Image{
+									Registry:   "gcr.io/gloo-mesh",
+									Repository: "gloo-mesh-mgmt-server",
+									Tag:        "0.0.1",
+								},
+								VolumeMounts: []v1.VolumeMount{{
+									Name:      "license-keys",
+									MountPath: "/etc/gloo-mesh/license-keys",
+									ReadOnly:  true,
+								}},
+							},
+							Volumes: []v1.Volume{
+								{
+									Name: "license-keys",
+									VolumeSource: v1.VolumeSource{
+										Secret: &v1.SecretVolumeSource{
+											SecretName: "license-keys",
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						Name:                  "gloo-agent",
+						CustomEnableCondition: `and ($.Values.glooAgent.enabled) (not $.Values.glooAgent.runAsSidecar)`,
+					},
+				},
+			},
+			ManifestRoot: "codegen/test/chart",
+		}
+
+		Expect(cmd.Execute()).NotTo(HaveOccurred(), "failed to execute command")
+
+		absPath, err := filepath.Abs("./test/chart/templates/deployment.yaml")
```

Comment:
This path is used in another test, We are not really seeing these templates for review here as they are getting overwritten by a later test